server/autotest_server/testers/ai/ai_tester.py

```diff
@@ -87,13 +88,23 @@ def call_ai_feedback(self) -> dict:
         try:
             result = subprocess.run(cmd, capture_output=True, text=True, check=True, timeout=timeout, env=env)
             output = result.stdout
+
+            parsed = None
+            try:
+                parsed = json.loads(output)
+            except json.JSONDecodeError:
+                pass
+            if isinstance(parsed, dict) and "tags" in parsed:
+                tags = parsed.pop("tags")
+                output = parsed.pop("output")
+                self.tags.extend(tags)
+
             if output_mode == "overall_comment":
                 self.overall_comments.append(output)
                 results[test_label] = {"title": test_label, "status": "success"}
             elif output_mode == "annotations":
                 try:
-                    annotations_data = json.loads(output)
-                    annotations = annotations_data.get("annotations", annotations_data)
+                    annotations = parsed.get("annotations", parsed)
```

Comment:
We should change the surrounding code as well. If `parsed is None` we can raise a `ValueError` (note that with your changes, a `json.JSONDecodeError` will never be thrown here)